### PR TITLE
add metricrangedf and tests

### PR DESCRIPTION
**Pull request by allenmqcymp**
Hi @4n4nd , @chauhankaranraj , I created this new branch because I had some messy git business in my other branch when I tried to **rebase**. Rather than fixing the cryptic errors, I just made this new branch and copied all the code over, hope that's OK. I'll delete the other branch.

### CODE REVIEW

**Comment by 4n4nd** [comment]
@allenmqcymp looks like some formatting issues crept in

```
prometheus_api_client/__init__.py
|  10| from·.metric_range_df·import·MetricRangeDataFrame
|    | [NORMAL] PycodestyleBear (W292):
|    | W292 no newline at end of file
[33m[WARNING][0m[15:39:57] PycodestyleBear (W292): This result has no patch attached.[0m

prometheus_api_client/metric_range_df.py
|  77| ········self.set_index(["timestamp"],·inplace=True)
|    | [NORMAL] PycodestyleBear (W292):
|    | W292 no newline at end of file
```

**Comment by allenmqcymp** [comment]
@4n4nd , fixed and rebased

**Comment by 4n4nd** [comment]
retest

**Comment by 4n4nd** [comment]
recheck

**Comment by allenmqcymp** [comment]
OK this issue is extremely puzzling and annoying for me. My local and origin/metricrange2 are synced together. 

git status
> On branch metricrange2
> Your branch is up to date with 'origin/metricrange2'.

When I git push it says "everything up to date" - so my origin and local code should look the same right? Well my editor (VSCode) shows that the files which need newlines have them, yet on github's viewer, they apparently do not....which is throwing the thoth-coala error. What is going on here?

**Comment by chauhankaranraj** [comment]
> When I git push it says "everything up to date" - so my origin and local code should look the same right? Well my editor (VSCode) shows that the files which need newlines have them, yet on github's viewer, they apparently do not....which is throwing the thoth-coala error. What is going on here?

Hm that's weird. When I do the following steps, I do not see a newline at the EOF.
```
1. git clone https://github.com/allenmqcymp/prometheus-api-client-python.git
2. git fetch && git checkout metricrange2
3. vim prometheus_api_client/metric_range_df.py
```
Then when add a newline using `vim` and run `git status` again, I am able to see the diff (see outputs below)
```
git status
On branch metricrange2
Your branch is up to date with 'origin/metricrange2'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   prometheus_api_client/metric_range_df.py

no changes added to commit (use "git add" and/or "git commit -a")
```
```
git diff metricrange2
diff --git a/prometheus_api_client/metric_range_df.py b/prometheus_api_client/metric_range_df.py
index deee05c..d6c6b96 100644
--- a/prometheus_api_client/metric_range_df.py
+++ b/prometheus_api_client/metric_range_df.py
@@ -75,3 +75,4 @@ class MetricRangeDataFrame(DataFrame):
         )
 
         self.set_index(["timestamp"], inplace=True)
+
(END)
```
Is it possible that your VSCode settings are configured to remove newlines? Or perhaps your git settings are configured to ignore newline changes?

**Comment by allenmqcymp** [comment]
> > When I git push it says "everything up to date" - so my origin and local code should look the same right? Well my editor (VSCode) shows that the files which need newlines have them, yet on github's viewer, they apparently do not....which is throwing the thoth-coala error. What is going on here?
> 
> Hm that's weird. When I do the following steps, I do not see a newline at the EOF.
> 
> ```
> 1. git clone https://github.com/allenmqcymp/prometheus-api-client-python.git
> 2. git fetch && git checkout metricrange2
> 3. vim prometheus_api_client/metric_range_df.py
> ```
> 
> Then when add a newline using `vim` and run `git status` again, I am able to see the diff (see outputs below)
> 
> ```
> git status
> On branch metricrange2
> Your branch is up to date with 'origin/metricrange2'.
> 
> Changes not staged for commit:
>   (use "git add <file>..." to update what will be committed)
>   (use "git restore <file>..." to discard changes in working directory)
> 	modified:   prometheus_api_client/metric_range_df.py
> 
> no changes added to commit (use "git add" and/or "git commit -a")
> ```
> 
> ```
> git diff metricrange2
> diff --git a/prometheus_api_client/metric_range_df.py b/prometheus_api_client/metric_range_df.py
> index deee05c..d6c6b96 100644
> --- a/prometheus_api_client/metric_range_df.py
> +++ b/prometheus_api_client/metric_range_df.py
> @@ -75,3 +75,4 @@ class MetricRangeDataFrame(DataFrame):
>          )
>  
>          self.set_index(["timestamp"], inplace=True)
> +
> (END)
> ```
> 
> Is it possible that your VSCode settings are configured to remove newlines? Or perhaps your git settings are configured to ignore newline changes?

I tried using Vim and the diff shows, great. But when I pushed to github it still doesn't show, leading me to think it's github ignoring the trailing newlines.

**Comment by 4n4nd** [comment]
regate

**Comment by 4n4nd** [comment]
recheck

**Comment by 4n4nd** [comment]
@chauhankaranraj looks like it passed the test, could you please approve it?
Also, could you please squash it before merging?

Thanks @chauhankaranraj and @allenmqcymp :+1:

File: tests/test_metric_range_df.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        # check shape when list with single json passed
+        self.assertEqual(
+            (num_vals, 5),
+            MetricRangeDataFrame([self.raw_metrics_list[0][0]]).shape,
+            "incorrect dataframe shape when initialized with single json list",
+        )
+
+
+if __name__ == "__main__":
+    unittest.main()
```
Comment (martinpovolny) [inline tests/test_metric_range_df.py:97]: Newline missing at the end of the file.

Most editors have an option to fix this for you by default.

Comment (4n4nd) [reply tests/test_metric_range_df.py:97]: let's fix that in the next PR

**Comment by 4n4nd** [comment]
@martinpovolny is it okay if we merge this PR?

File: tests/test_metric_range_df.py (hunk 1 above)
Comment (martinpovolny) [reply tests/test_metric_range_df.py:97]: sure, can be done in a new PR

**Comment by martinpovolny** [comment]
I am more concerned about the repetition of the code that loads the fixtures. (addressed in https://github.com/AICoE/prometheus-api-client-python/pull/140), but never mind, I can extend my PR to fix that in this new file too.

**Comment by martinpovolny** [comment]
> @martinpovolny is it okay if we merge this PR?

Sure, go ahead.

**Review by Gdakgdak22 — APPROVED** [review]

**Pull request merged**